dev/source/docs/ros2-pi.rst

```diff
@@ -95,3 +95,183 @@ The simplest solution is to wire an open telemetry port.
+
+..
+    Inspired by https://github.com/Ryanf55/ardupilot_ros/blob/docker-deploy-to-pi/docker/Dockerfile
+
+In order to run applications on a Raspberry Pi, we must cross compile them for the ARM architecture.
+`Docker <https://www.docker.com/>`_ allows an easy way to cross compile. Because of the Pi's limited memory, 
+it is not recommended to attempt compiling on target hardware.
+
+First, create a dockerfile like so on your laptop called ``Dockerfile``
+
+.. code-block:: docker
+
+    FROM arm64v8/ros:humble
+
+    RUN --mount=target=/var/lib/apt/lists,type=cache,sharing=locked \
+        --mount=target=/var/cache/apt,type=cache,sharing=locked \
+        apt-get update && apt-get -y --no-install-recommends install \
+            git cmake build-essential
+
+    WORKDIR /ws
+    RUN git clone  --depth 1 --branch humble https://github.com/micro-ROS/micro-ROS-Agent.git
+    WORKDIR /ws/micro-ROS-Agent
+    RUN --mount=target=/var/lib/apt/lists,type=cache,sharing=locked \
+        --mount=target=/var/cache/apt,type=cache,sharing=locked \
+        apt-get update && \
+        rosdep update && \
+        rosdep install --from-paths . --ignore-src -y && \
+        apt-get -y --no-install-recommends install ros-humble-fastcdr
+
+    RUN . /opt/ros/humble/setup.sh && colcon build
+
+Next, use Docker's multi-platform build support to set up a cross compile in an ARM64v8 environment: https://docs.docker.com/build/building/multi-platform/#install-qemu-manually
+
+.. code-block:: bash
+
+    docker run --privileged --rm tonistiigi/binfmt --install all
+    cat /proc/sys/fs/binfmt_misc/qemu-arm | grep flags
+
+Now, the following steps build the docker image and copy the ROS 2 install directory to the target.
+
+.. code-block:: bash
+
+    # Build with multiplatform support.
+    docker build . --platform linux/arm64 -t micro_ros_agent
+
+    # Copy the installation directory from the docker image to the host.
+    docker create --name dummy micro_ros_agent
+    docker cp dummy:/ws/micro-ROS-Agent/install install
+    docker rm -f dummy
+
+    # And now, from host to target's home directory, but ignore COLCON_IGNORE.
+    rsync -aRv --exclude install/COLCON_IGNORE install ubuntu@ubuntu.local:/home/ubuntu
+
+
+Although we have copied the executable over, the runtime dependencies are not installed.
+Let's use rosdep to install those by cloning it on the target and invoking rosdep.
+The target only needs runtime dependencies, which are denoted with tag ``exec``.
+
+.. code-block:: bash
+
+    cd ~
+    sudo rosdep init
+    apt update
+    rosdep update
+    rosdep install --from-paths install --dependency-types exec
+
+Test the Micro ROS Agent
+========================
+
+Now that you have a cross-compiled installation of the Micro ROS Agent along with its dependencies, let's check we can run it.
+
+.. code-block:: bash
+
+    source ~/install/setup.bash
+    ros2 run micro_ros_agent micro_ros_agent serial -v4 -b 115200 -D /dev/ttyS0
+
```

Comment:
Also add in a section about testing by subscribing to topics on the remote laptop and local device

Reply:
Added this in the "coming soon" section. Good idea to remind people about the CLI.